src/java/org/apache/cassandra/db/tries/Trie.java

```diff
@@ -275,7 +275,7 @@ interface Walker<T, R> extends ResettingTransitionsReceiver
     }
 
     // Version of the byte comparable conversion to use for all operations
-    static final ByteComparable.Version BYTE_COMPARABLE_VERSION = ByteComparable.Version.OSS41;
+    static final ByteComparable.Version BYTE_COMPARABLE_VERSION = ByteComparable.Version.OSS41; // TODO hardcoded encoding version
```

Comment:
The thing is, it is hard to enforce this is used only by in-memory structures. I'd rather make in-memory structures pass a hardcoded version instead. Though, this would be address in the other ticket, @pkolaczk is working on. Let me close this PR as a fix just for SSTables, I can move todos to the other ticket

Reply:
Actually, this is not public so enforcing not being used outside of in-memory structures is only down to making sure it stays package-private. I would add a comment (to not change to public and to not use in on-disk structures) and otherwise leave this part unchanged.